### ISSUE TITLE
TorchServe KeyError

### ISSUE THREAD

**Issue by austinmw** [open-mmlab/mmdetection#9301]
### Discussed in https://github.com/open-mmlab/mmdetection/discussions/9200

<div type='discussions-op-text'>

<sup>Originally posted by **austinmw** October 31, 2022</sup>
Hi, I used MMDetection 3.0.0rc2 to train a Deformable DETR model and a Faster RCNN model, and then I used the mmdet2torchserve tool to create two `model.mar`. I tried testing local deployment with my own TorchServe Dockerfile, but I get errors with both of the TorchServe archives:

> KeyError: "class `DeformableDETR` in mmdet/models/detectors/deformable_detr.py: 'DetDataPreprocessor is not in the model registry. Please check whether the value of `DetDataPreprocessor` is correct or it was registered as expected.

and 

> KeyError: "class `FasterRCNN` in mmdet/models/detectors/faster_rcnn.py: 'DetDataPreprocessor is not in the model registry. Please check whether the value of `DetDataPreprocessor` is correct or it was registered as expected.

My Dockerfile looks like this:

```Dockerfile
# Use Sagemaker PyTorch inference container as base image
ARG REGION
FROM 763104351884.dkr.ecr.${REGION}.amazonaws.com/pytorch-inference:1.12.1-gpu-py38-cu113-ubuntu20.04-sagemaker

# Install MMEngine, MMCV, MMDetection
WORKDIR /opt/ml/code
RUN pip install cython nvgpu && \
    pip install -U pip openmim torch-model-archiver && \
    pip install git+https://github.com/open-mmlab/mmengine.git@v0.2.0 && \
    mim install "mmcv>=2.0.0rc2" -f https://download.openmmlab.com/mmcv/dist/cu113/torch1.12.0/index.html && \
    pip install git+https://github.com/open-mmlab/mmdetection.git@v3.0.0rc2 && \
    rm -rf /var/lib/apt/lists/*

COPY config.properties /home/model-server/config.properties

WORKDIR /
ENTRYPOINT ["python", "/usr/local/bin/dockerd-entrypoint.py"]
CMD ["torchserve", "--start", "--ts-config", "/home/model-server/config.properties", "--model-store", "/opt/ml/model"]
```
I placed the `model.mar`s into `/opt/ml/model`.

Anyone know how to resolve this KeyError? Is `mmdet_handler.py` compatible with `v3.0.0rc2`?
</div>

**Comment by BIGWangYuDong** [comment]
Hi, sorry for the late reply. scripts in `tools/deployment` did not compatible with 3.0 yet. This may occur some errors now. It is pleased to create PR to compatible with 3.0.

**Comment by austinmw** [comment]
Hi, thanks for response, okay I'll see if I can get it working for PR, otherwise can wait till official 3.0 release

**Comment by ZwwWayne** [comment]
@lvhan028 to take look. MMDeploy should already be compatible with mmdet 3.x in mmdeploy 1.x. We just need to have an entry point in mmdet.

**Comment by RunningLeon** [comment]
> @lvhan028 to take look. MMDeploy should already be compatible with mmdet 3.x in mmdeploy 1.x. We just need to have an entry point in mmdet.

Hi, torchserve is not included in mmdeploy. Deployment for torchserve would be kept in mmdet.
As for the error, it's because of the BC-breaking of mmdet2.x and mmdet3.x. A fixing of this would come later.

**Comment by ZwwWayne** [comment]
Resolved in #9343
